Mathlib/Util/PPOptions.lean

```diff
@@ -29,3 +29,17 @@ def getPPBinderPredicates (o : Options) : Bool :=
   o.get pp.mathlib.binderPredicates.name (!getPPAll o)
 
 end Mathlib
+
+
+/--
+The `pp.mdata` option is used to control whether `Expr.mdata` nodes are pretty-printed.`
+-/
+register_option pp.mdata : Bool := {
+  defValue := false
+  group    := "pp"
+  descr    := "(pretty printer) pretty prints mdata annotations"
+}
+
+/-- Gets whether `Expr.mdata` delaboration is enabled. -/
+def Lean.getPPMData (o : Options) : Bool :=
+  o.get pp.mdata.name (getPPAll o)
```

Comment:
I don't know about having `pp.all` show mdata for general mathlib users. Would you mind making this need to be enabled on purpose?
```suggestion
  o.get pp.mdata.name pp.mdata.default
```